torch/testing/_internal/distributed/rpc/jit/rpc_test.py

```diff
@@ -1116,6 +1127,40 @@ def test_rpc_async_jit_profiled(self):
             remote_add_profiled_name = f"{profiled_name}#remote_op: aten::add"
             self.assertEqual(remote_add.name, remote_add_profiled_name)
 
+    @dist_init
+    def test_rpc_torchscript_record_function(self):
```

Comment:
Can we add a test where `with record_function` is on the caller side inside a torchscript function and we call `rpc_async` within the `record_function` context manager?

Reply:
@pritamdamania87 Just to make sure we're on the same page, were you thinking of something like the following?
```
@torch.jit.script
def foo():
    with record_function(name) as rf:
        rpc.rpc_async(....)
        rf._call_end_callbacks_on_future
```

and ensure we have the correct time for `rpc_async`?

Reply:
@pritamdamania87 Actually it doesn't seem like the above test will work unless the return future of `rpc_async` in JIT is typed as `Future[Any]`, due to current TorchScript limitations. Although, I don't anticipate a use case such as the above where a user manually has to call `rf._call_end_callbacks_on_future`, that should be a Python only thing which is called internally. For rpc_async in JIT, the profiling is done in the C++ operator. Or did you have a different idea of what this test should look like?

Reply:
@rohan-varma I meant something similar to your example, but more concretely:

```
@torch.jit.script
def foo():
    with record_function(name) as rf:
        fut1 = rpc.rpc_async(....)
        fut2 = rpc.rpc_async(....)
        fut1.wait()
        fut2.wait()
        res = fut1.value() + fut2.value()
```

I'm guessing this should work since we're not actually using `rf._call_end_callbacks_on_future`?